connectors/sources/mysql.py

```diff
@@ -108,6 +109,71 @@ async def _remote_validation(self, advanced_rules):
         )
 
 
+class MySQLClient:
+    def __init__(
+        self,
+        host,
+        port,
+        user,
+        password,
+        ssl_enabled,
+        ssl_certificate,
+        db=None,
+        max_pool_size=MAX_POOL_SIZE,
+    ):
+        self.host = host
+        self.port = port
+        self.user = user
+        self.password = password
+        self.db = db
+        self.max_pool_size = max_pool_size
+        self.ssl_enabled = ssl_enabled
+        self.ssl_certificate = ssl_certificate
+
+    @asynccontextmanager
+    async def with_connection_pool(self):
+        connection_string = {
+            "host": self.host,
+            "port": int(self.port),
+            "user": self.user,
+            "password": self.password,
+            "db": self.db,
+            "maxsize": self.max_pool_size,
+            "ssl": _ssl_context(certificate=self.ssl_certificate)
+            if self.ssl_enabled
+            else None,
+        }
+
+        connection_pool = None
+
+        try:
+            connection_pool = await aiomysql.create_pool(**connection_string)
+            yield connection_pool
+        except Exception:
+            logger.error("Failed to create connection pool for MySQL.")
+            raise
+        finally:
+            connection_pool.close()
+            await connection_pool.wait_closed()
+
+
+def _ssl_context(certificate):
```

Comment:
why not reusign `connectors.utils.ssl_context`

Reply:
Wasn't aware that `_ssl_context` was duplicated in the first place. Will fix that

Reply:
Adjusted it, thank you for the hint 👍